src/serialbox-c/FortranWrapper.cpp

```diff
@@ -25,26 +25,47 @@ using namespace serialboxC;
 using serialbox::Exception;
 using serialbox::TypeID;
 
+namespace {
+void make_4D(std::vector<int>& v) {
+  while(v.size() < 4) {
+    v.push_back(0);
+  }
+}
+
+std::vector<int> make_strides(int istride, int jstride, int kstride, int lstride) {
+  std::vector<int> strides;
+  if(istride >= 0)
+    strides.push_back(istride);
+  if(jstride >= 0)
+    strides.push_back(jstride);
+  if(kstride >= 0)
+    strides.push_back(kstride);
+  if(lstride >= 0)
+    strides.push_back(lstride);
```

Comment:
Can it be that `istride == 0`, but `kstride > 0`?

I suppose it can be when you have, say a 3D fortran field and you are passing it to serialbox like this:

```
field(3, :, :)
```

Sorry for being pedantic/paranoid, I'm just trying to make sure I understand the implications with respect to the 0/1 dimension size convention.

Reply:
Thanks for being pedantic. I think you cannot have this case if you use the `fs_write_XXX_Nd` or `fs_read_XXX_Nd` functions as they will always put the `-1`. I don't think `istride==0` is a problem.

In your example it will be passed as a 2D array (I had to ask Fortran guys here). If you pass a 3D array with size 1 in the first dimension, I think it will still work.